pinot-common/src/main/java/org/apache/pinot/common/utils/config/QueryOptionsUtils.java

```diff
@@ -166,4 +166,28 @@ public static boolean isServerReturnFinalResult(Map<String, String> queryOptions
   public static String getOrderByAlgorithm(Map<String, String> queryOptions) {
     return queryOptions.get(QueryOptionKey.ORDER_BY_ALGORITHM);
   }
+
+  @Nullable
+  public static Integer getMultiStageLeafLimit(Map<String, String> queryOptions) {
+    String maxLeafLimitStr = queryOptions.get(QueryOptionKey.MULTI_STAGE_LEAF_LIMIT);
+    return maxLeafLimitStr != null ? Integer.parseInt(maxLeafLimitStr) : null;
+  }
+
+  @Nullable
+  public static Integer getNumGroupLimit(Map<String, String> queryOptions) {
```

Comment:
Suggest keeping the name consistent for readability, same for other getters
```suggestion
  public static Integer getNumGroupsLimit(Map<String, String> queryOptions) {
```